src/main/java/com/blamejared/crafttweaker/impl/recipes/handlers/crafttweaker/CTShapelessRecipeHandler.java

```diff
@@ -0,0 +1,42 @@
+package com.blamejared.crafttweaker.impl.recipes.handlers.crafttweaker;
+
+import com.blamejared.crafttweaker.api.item.IIngredient;
+import com.blamejared.crafttweaker.api.managers.IRecipeManager;
+import com.blamejared.crafttweaker.api.recipes.IRecipeHandler;
+import com.blamejared.crafttweaker.api.recipes.IReplacementRule;
+import com.blamejared.crafttweaker.api.recipes.ReplacementHandlerHelper;
+import com.blamejared.crafttweaker.api.util.StringUtils;
+import com.blamejared.crafttweaker.impl.item.MCItemStackMutable;
+import com.blamejared.crafttweaker.impl.recipes.CTRecipeShapeless;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@IRecipeHandler.For(CTRecipeShapeless.class)
+public final class CTShapelessRecipeHandler implements IRecipeHandler<CTRecipeShapeless> {
+    
+    @Override
+    public String dumpToCommandString(final IRecipeManager manager, final CTRecipeShapeless recipe) {
+    
+        return String.format(
+                "craftingTable.addShapeless(%s, %s, %s);",
```

Comment:
Should we print something if the recipe has a function?

Reply:
Added a print for this case. The format is still up for discussion.